api/src/db/migrate.ts

```diff
@@ -0,0 +1,259 @@
+import Session from '../models/session';
+import { db } from '.';
+import { planner, report, review, savedCourse, session, transferredCourse, user, vote } from './schema';
+import Review from '../models/review';
+import Roadmap from '../models/roadmap';
+import Preference from '../models/preference';
+import Report from '../models/report';
+import Vote from '../models/vote';
+import { anonymousName, MongoRoadmap, SavedPlannerData, SavedPlannerYearData } from '@peterportal/types';
+import mongoose from 'mongoose';
+
+const uri = process.env.MONGO_URL;
+const conn = await mongoose.connect(uri!, {
+  dbName: 'peterPortalDevDB',
+  serverSelectionTimeoutMS: 5000,
+});
+
+// clear first
+await Promise.all([
+  db.delete(planner),
+  db.delete(vote),
+  db.delete(report),
+  db.delete(review),
+  db.delete(savedCourse),
+  db.delete(transferredCourse),
+  db.delete(session),
+  db.delete(review),
+  db.delete(user),
+]);
+
+// get all possible users from sessions, then reviews, then roadmaps, then votes
+const sessions = await Session.find();
+// need to filter out bad sessions (never signed in with google, just clicked log in)
+const sessionsUserRecords = sessions
+  .filter((session) => session.session?.passport?.user?.id != null)
+  .map((session) => ({
+    googleId: session.session!.passport!.user!.id!,
+    name: session.session!.passport!.user!.name,
+    email: session.session!.passport!.user!.email,
+    picture: session.session!.passport!.user!.picture!,
+  }));
+
+const userIdMapping: Record<string, number> = {};
+
+const newIdsFromSessions = await db
+  .insert(user)
+  .values(sessionsUserRecords)
+  .onConflictDoNothing()
+  .returning({ id: user.id, googleId: user.googleId });
+for (const newId of newIdsFromSessions) {
+  userIdMapping[newId.googleId] = newId.id;
+}
+
+const reviewDocs = await Review.find();
+for (const review of reviewDocs) {
+  const newId = await db
+    .insert(user)
+    .values({ googleId: review.userID, name: review.userDisplay, email: '', picture: '' })
+    .onConflictDoNothing()
+    .returning({ id: user.id });
+  if (newId.length > 0) {
+    userIdMapping[review.userID] = newId[0].id;
+  }
+}
+
+const roadmaps = await Roadmap.find<Omit<MongoRoadmap, 'userId'> & { userID: string }>();
+for (const roadmap of roadmaps) {
+  const newId = await db
+    .insert(user)
+    .values({
+      googleId: roadmap.userID,
+      name: anonymousName,
+      email: '',
+      picture: '',
+      ...(roadmap.roadmap.timestamp ? { lastRoadmapEditAt: new Date(Number(roadmap.roadmap.timestamp)) } : {}),
```

Comment:
similar to dates the other way, we should have a helper to construct this pattern
`dateLookup('lastRoadmapEditAt', roadmap.roadmap.timestamp)`